aiohttp/multipart.py

```diff
@@ -310,13 +311,14 @@ def _read_chunk_from_stream(self, size):
             self._prev_chunk = yield from self._content.read(size)
 
         chunk = yield from self._content.read(size)
-
+        self._content_eof += int(self._content.at_eof())
+        assert self._content_eof < 3, "Reading after EOF"
```

Comment:
I think whole method could be simplified if there will be `._content.readline(size: int=N)` method that reads whole line or at least N bytes. Then all what would be need is correctly handle boundary that split across multiple chunks. Something like:
```
def _read_chunk_from_stream(self, size):
  chunk = yield from self._content.maybe_readline(size=size)
  if chunk == self._boundary: 
    self._at_eof = True
    return b''
  if self._boundary.startswith(chunk): 
    return (yield from maybe_finish_boundary_read(chunk))  # should handle false positive match
  return chunk
```